lib/main.dart

```diff
@@ -1,22 +1,31 @@
+import "package:firebase_core/firebase_core.dart";
 import "package:flutter/material.dart";
 import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/utils/firebase/firebase_options.dart";
 import "package:proxima/views/navigation/routes.dart";
 
-void main() {
-  runApp(const MyApp());
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+
+  runApp(
+    const ProviderScope(
+      child: ProximaApp(),
+    ),
+  );
 }
 
```

Comment:
Maybe we could add a global listener on the `userProvider` to listen to the authentication state and redirect toward the login screen if the user is disconnected.
This way, we can handle the case where the user is unexpectidly disconnected from an exterior source (for example if the user goes in his google account and select "disconnect from all devices").

Reply:
Yes great suggestion! I added this provider onto the root widget of the application (`HomePage`) as it should never be popped from the widget stack, and we must attach this listener to a widget.

Check it out and let me know if you have further remarks.